### PR TITLE
Add support for automatic choice of Trilinos version.

### PR DESCRIPTION
**Pull request by Rombur**
I have tried ```AUTO``` and to set ```TRILINOS_VERSION``` on Ubuntu 15 and it works but I realized that we don't support Ubuntu 12. @tjhei could you add it, I don't have access to Ubuntu 12.

### CODE REVIEW

**Comment by koecher** [comment]
@Rombur have you tried out vagrant for getting a minimal ubuntu12?
Pretty cool thing: https://www.vagrantup.com

**Comment by tjhei** [comment]
can you cherry-pick my commit that adds the ubuntu12 package? https://github.com/tjhei/candi/commit/a8bfea713e603489e565240d88b1cfcb152ce28c

File: candi.sh [hunk 1]
```diff
@@ -706,6 +706,15 @@ guess_architecture
 # Reset timings
 TIMINGS=""
 
+# If TRILINOS_MAJOR_VERSION is set to AUTO, pick Trilinos 11 except for Ubuntu 12
+if [[ ${TRILINOS_MAJOR_VERSION} == "AUTO" ]]; then
```
Comment (tjhei) [inline candi.sh]: Can you move this check into ubuntu12.platform? (I have no idea if it is executed early enough)

Comment (Rombur) [inline candi.sh]: This is executed once the platform is determined but before we start installing the packages.

**Comment by Rombur** [comment]
@koecher never heard of vagrant. It is seems really nice.

File: candi.sh (hunk 1 above)
Comment (tjhei) [inline candi.sh]: yes, but wouldn't it make a lot more sense to make the decision "if ubuntu12 use Trilinos11 not 12" inside ubuntu12.platform?

Comment (Rombur) [inline candi.sh]: yeah that would work too. I am going to steal your ubuntu12 file and move it there :-)

**Pull request merged**